app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenter.java

```diff
@@ -20,8 +21,16 @@ public void onShowButtonPressed(ListenerDialogFragment dialogFragmentListener) {
 
     @Override
     public void listenParkingAvailable(String parkingAvailable) {
-        model.setParkingAvailable(Integer.parseInt(parkingAvailable));
-        view.showNumberOfParkingAvailable(String.valueOf(model.getParkingAvailable()));
+        model.setParkingLots(parkingAvailable);
+        view.showNumberOfParkingAvailable(String.valueOf(model.getParkingLots()));
     }
 
+    @Override
+    public void onReservationButtonPressed() {
+        if (Integer.parseInt(model.getParkingLots()) > Constants.DEFAULT_PARKING){
```

Comment:
what happens if getParkingLots return 0?

Reply:
whose is the responsibility of this validation?

Reply:
It won't return 0, because in the presenter in the dialog fragment (main activity) it checks that if the value of the parking is 0, the view shows a toast that saying that the number is not valid, that is has to be minimum 1.